grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1381,6 +1387,35 @@ class JsonController {
         render(["aaData": data] as JSON)
     }
 
+    def getSummaryOrderReport = {
+        def location = Location.get(session.warehouse.id)
+
+        String query = """
+            select * 
+            from on_order_summary 
+            where destination_id = :locationId
+            """
+        Sql sql = new Sql(dataSource)
+        def items = sql.rows(query,  [locationId: location.id])
+
+        def data = items.collect {
+            Product product = Product.findByProductCode(it[0].toString())
```

Comment:
This is going to require N queries where N is the number of products. You could probably do this more efficiently by collecting all IDs outside the loop and then doing a lookup in the 
```
def productIds = items.collect { it.productId }
def productsByProductCode = Product.getAll(productIds).groupBy { it.productCode }
def data = items.collect {
    Product product = productsByProductCode.get(it[0])
}
```
OR better yet. Include the product name in the query.